Scripts/summaryStatistics/biophysics/growth_rate_analysis.py

```diff
@@ -0,0 +1,247 @@
+    
+    @param  cell    cellStates object
+    @param  centerX X-coordinate of colony centroid
+    @param  centerY Y-coordinate of colony centroid
+    @return r       Radial position of cell relative to colony centroid
+    """
+    
+    r = math.sqrt((cell.pos[0] - centerX)**2 + (cell.pos[1] - centerY)**2)
+    return r
+    
+def define_bin_ranges(bin_radius, r):
+    """
+    Define the radial ranges for bins to group cells by.
+    
+    @param  bin_radius  Length of the annuli to group cells by
+    @param  r           Radial distance from centroid (nparray)
+    @return bin_ranges  Bin intervals (distances) to average over (nparray)
+    """
+    # Get upper bound for the colony radius
+    max_r = round(np.max(r))
+    
+    # Creat bin ranges
+    bin_ranges = np.arange(0, math.ceil(max_r), bin_radius) #0-10, 10-20, etc.
+    
+    return bin_ranges
+    
+def bin_by_radius(r, measurement, bin_radius, centerX, centerY):
+    """
+    Place cell measurements into bins based on radial position in a colony
+    
+    @param  r           nparray of radial distances from colony centroid
+    @param  measurement measurement of interest (float)
+    @param  bin_radius  Thickness of band to average over (um)
+    @param  centerX     Center of colony (X-coord)
+    @param  centerY     Center of colony (Y-coord)
+    @return df          dataFrame containing: ['Radius'], ['Measurement'], ['bin']
+    """ 
+    # Assign bins to each r and measurement
+    bin_ranges = define_bin_ranges(bin_radius, r)
+    bin_indices = np.digitize(r, bin_ranges) #Allocate each r to a bin number; first bin = 1 #https://www.adamsmith.haus/python/answers/how-to-put-data-into-bins-using-numpy-in-python
+    
+    # Create a dataFrame of radial position, measurement, and bin index
+    df = pd.DataFrame()
+    df['Radius'] = r.tolist()
+    df['Growth Rate'] = measurement.tolist()
+    df['bin'] = bin_indices.tolist()
+    
+    return df
+    
+def get_binned_statistics(df):
+    """
+    Calculate the mean and standard deviation of binned data
+    
+    @param  df          dataFrame with a column ['bin']; the column contains integers
+    @return mean, std   Group dataFrames: mean and standard deviation of measurements (float)
+    """
+    # Group each cell by bin
+    df_groupby_bin = df.groupby(df['bin'])
+    
+    # Calculate statistics for all other measurements
+    mean = df_groupby_bin.mean()
+    std = df_groupby_bin.std()
+    
+    return mean, std
+    
+def filter_by_cell_age(cells, min_age=2):
+    """
+    Create a dict with cells that have age >= min_age
+    
+    @param  cells               cellStates dict
+    @param  min_age             minimum age to include cell in the filtered list; recommended age is 2
+    @return filtered_cellstates cellStates dict with cells filtered by age
+    """
+    filtered_cellstates = {}
+    for i, cell in cells.items():
+        if cell.cellAge >= min_age:
+            filtered_cellstates[i] = cell
```

Comment:
It is better to use this coding style. It is much faster and more professional and clear:

filtered_cellstates = {cell for cell in cells.items() if cell.cellAge >= min_age}

Reply:
Roger.

Reply:
I tried implementing your solution but I cannot get it to save both the key and value, so I get an error when I run the code. Here's what I tried:
    filtered_cellstates = {cell for (i, cell) in cells.items() if cell.cellAge >= min_age}

I will leave as is for now since the code works, and we can update in a future commit once we figure out how to get the more elegant style working.